File: backend/src/main/kotlin/com/example/backend/models/SortieStock.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "sortie_stock")
class SortieStock {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @ManyToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "en_rayon_id")
  var enRayon: EnRayon? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "type_sortie_id")
  var typeSortie: com.example.backend.models.TypeSortie? = null

  @Column(name = "quantite")
  var quantite: Int? = null

  @Column(name = "dateSortie")
  var dateSortie: LocalDateTime? = null

  @Column(name = "detail_id", length = 20)
  var detailId: String? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var  supprimer: Int? = 0
}

File: backend_py/pharmaPython/app/models/sortie_stock.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base

class SortieStock(Base):
  __tablename__ = "sortie_stock"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True, autoincrement=True)
  en_rayon_id = Column(Integer, ForeignKey("en_rayon.id"))
  type_sortie_id = Column(Integer, ForeignKey("type_sortie.id"))
  quantite = Column(Integer)
  date_sortie = Column(DateTime)
  detail_id = Column(String(20))
  supprimer = Column(Integer, default=0)

  en_rayon = relationship("EnRayon")
  type_sortie = relationship("TypeSortie")
